### DISCUSSION TITLE
profiling build fails

### DISCUSSION THREAD

**Discussion by heinono1** [idaholab/moose#25260] [Q&A General] [answered]
I am trying to build MOOSE with profiling (gperftools) enabled. I install the gperftools, build petsc and libmesh with the appropriate flags (in particular for libmesh METHOD=oprof). I build MOOSE with METHOD=oprof but the build fails when it comes to the linker stage: 
Linking Library /home/olle/projects2/moose/framework/contrib/pcre/libpcre-oprof.la...
/usr/bin/ld: cannot find -lmesh_oprof: No such file or directory
/usr/bin/ld: cannot find -ltimpi_oprof: No such file or directory
collect2: error: ld returned 1 exit status
make: *** [/home/olle/projects2/moose/framework/moose.mk:371: /home/olle/projects2/moose/framework/contrib/pcre/libpcre-oprof.la] Error 1
It seems to me the libmesh_oprof and libtimpi_oprof should be built and installed during the libmesh build, but those libraries simple are not there. 

Any help will be greatly appreciated!

**Comment by GiudGiud** [comment, marked as answer]
libmesh was not built with oprof seemingly.
I think libmesh obey METHODS not METHOD. Can you check the libmesh folder for what got built?

**Reply by heinono1** [reply]
Hi Guillaume. You're right that libmesh needs METHODS=oprof, not METHOD=oprof. Maybe not a bad idea to update the profiling build instructions.

Thanks!

**Reply by milljm** [reply]
Those instructions are a bit confusing. They don't go to mention how to build libMesh. Unless its assumed the user is using Pre-built Conda moose-libmesh? But then again, they go to mention how to build PETSc, and if you're building PETSc, then you must build libMesh immediately afterwards.

Reading those instructions again, I think the user is supposed to skip most of them, and just use `moose-pprof` package as part of the standard MOOSE installation instructions... ?

**Reply by lindsayad** [reply]
No. They are meant for people who are building the stack manually. Yes it would be good to explicitly mention how to build libMesh. The directions, when written, assumed that the users did not specify `METHODS` in their environment, so when they run the `scripts/update_and_rebuild_libmesh.sh` script they get oprof automatically (along with opt, devel, and dbg).

**Reply by lindsayad** [reply]
I've opened https://github.com/idaholab/moose/pull/25261. See if that is clearer
